src/components/SingleArticleCard.jsx

```diff
@@ -1,14 +1,18 @@
 import { useState, useEffect } from "react"
-import { getArticleById } from "../../api"
+import { getArticleById, patchVote } from "../../api"
 import { useParams } from "react-router-dom"
 import Comments from "./Comments"
 
 export default function SingleArticleCard(){
 
     const {article_id} = useParams()
     const [article, setArticle] = useState([])
-    // const [articleId, setarticleId] = useState("")
+    const [voteChange, setVoteChange] = useState(0)
 
+    const handleVote = (vote) => {
+        patchVote(article_id, vote);
```

Comment:
remember we will need to undo the optimistic render here if the vote fails to be added to the db